libraries/abstractions/transport/utest/transport_secure_sockets_utest.c

```diff
@@ -0,0 +1,776 @@
+                                                   NULL,
+                                                   &socketsConfig );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, returnStatus );
+
+    returnStatus = SecureSocketsTransport_Connect( &networkContext,
+                                                   &serverInfo,
+                                                   NULL );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, returnStatus );
+
+    invalidServerInfo.pHostName = NULL;
+
+    returnStatus = SecureSocketsTransport_Connect( &networkContext,
+                                                   &invalidServerInfo,
+                                                   &socketsConfig );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, returnStatus );
+
+    invalidServerInfo.pHostName = &hostNameBuffer[ 0 ];
+    invalidServerInfo.hostNameLength = 0;
+
+    returnStatus = SecureSocketsTransport_Connect( &networkContext,
+                                                   &invalidServerInfo,
+                                                   &socketsConfig );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, returnStatus );
+
+    for( index = 0U; index < INVALID_HOSTNAME_LENGTH; index++ )
+    {
+        hostNameBuffer[ index ] = '1';
+    }
+
+    invalidServerInfo.hostNameLength = strlen( hostNameBuffer );
+
+    SOCKETS_Close_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+
+    returnStatus = SecureSocketsTransport_Connect( &networkContext,
+                                                   &invalidServerInfo,
+                                                   &socketsConfig );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, returnStatus );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test that #SecureSocketsTransport_Connect fails on socket creation,
+ *  and verify the error code.
+ */
+void test_SecureSocketsTransport_Connect_Insufficient_Memory( void )
+{
+    TransportSocketStatus_t returnStatus;
+
+    SOCKETS_Socket_ExpectAnyArgsAndReturn( SOCKETS_INVALID_SOCKET );
+    SOCKETS_Close_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+    returnStatus = SecureSocketsTransport_Connect( &networkContext,
+                                                   &serverInfo,
+                                                   &socketsConfig );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INSUFFICIENT_MEMORY, returnStatus );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test that #SecureSocketsTransport_Connect fails, when setting
+ * the socket options to require TLS, and verify the error code.
+ */
+void test_SecureSocketsTransport_Connect_Invalid_Credentials_SetRequireTLS( void )
+{
+    TransportSocketStatus_t returnStatus;
+
+    SOCKETS_Socket_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
```

Comment:
We should make sure that the correct arguments are provided by the `transport_secure_sockets.c` implementation to the `SOCKETS_Socket` API with `SOCKETS_Socket_ExpectAndReturn(SOCKETS_AF_INET, SOCKETS_SOCK_STREAM, SOCKETS_IPPROTO_TCP, SOCKETS_ERROR_NONE );`

Reply:
fixed in e8830a7f9962ca0c9353a10fb1485eb6b8fb5e87